pkg/vault/contracts/AggregatorBatchRouter.sol

```diff
@@ -0,0 +1,476 @@
+                            sender: msg.sender,
+                            paths: paths,
+                            deadline: deadline,
+                            wethIsEth: wethIsEth,
+                            userData: userData
+                        })
+                    )
+                ),
+                (uint256[], address[], uint256[])
+            );
+    }
+
+    /// @inheritdoc IBatchRouter
+    function swapExactOut(
+        SwapPathExactAmountOut[] memory paths,
+        uint256 deadline,
+        bool wethIsEth,
+        bytes calldata userData
+    )
+        external
+        payable
+        saveSender(msg.sender)
+        returns (uint256[] memory pathAmountsIn, address[] memory tokensIn, uint256[] memory amountsIn)
+    {
+        return
+            abi.decode(
+                _vault.unlock(
+                    abi.encodeCall(
+                        AggregatorBatchRouter.swapExactOutHook,
+                        SwapExactOutHookParams({
+                            sender: msg.sender,
+                            paths: paths,
+                            deadline: deadline,
+                            wethIsEth: wethIsEth,
+                            userData: userData
+                        })
+                    )
+                ),
+                (uint256[], address[], uint256[])
+            );
+    }
+
+    function swapExactInHook(
+        SwapExactInHookParams calldata params
+    )
+        external
+        nonReentrant
+        onlyVault
+        returns (uint256[] memory pathAmountsOut, address[] memory tokensOut, uint256[] memory amountsOut)
+    {
+        (pathAmountsOut, tokensOut, amountsOut) = _swapExactInHook(params);
+
+        _settlePaths(params.sender, params.wethIsEth);
+    }
+
+    function _swapExactInHook(
+        SwapExactInHookParams calldata params
+    ) internal returns (uint256[] memory pathAmountsOut, address[] memory tokensOut, uint256[] memory amountsOut) {
+        // The deadline is timestamp-based: it should not be relied upon for sub-minute accuracy.
+        // solhint-disable-next-line not-rely-on-time
+        if (block.timestamp > params.deadline) {
+            revert SwapDeadline();
+        }
+
+        pathAmountsOut = _computePathAmountsOut(params);
+
+        // The hook writes current swap token and token amounts out.
+        // We copy that information to memory to return it before it is deleted during settlement.
+        tokensOut = _currentSwapTokensOut().values();
+        amountsOut = new uint256[](tokensOut.length);
+        for (uint256 i = 0; i < tokensOut.length; ++i) {
+            amountsOut[i] = _currentSwapTokenOutAmounts().tGet(tokensOut[i]);
+        }
+    }
+
+    function _computePathAmountsOut(
+        SwapExactInHookParams calldata params
+    ) internal returns (uint256[] memory pathAmountsOut) {
+        pathAmountsOut = new uint256[](params.paths.length);
+
+        if (EVMCallModeHelpers.isStaticCall() == false) {
```

Comment:
Should comment this block, as this is the different part compared to the regular BatchRouter.